pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -185,7 +173,7 @@ func (i *instantSplitter) mapBinaryExpr(expr *parser.BinaryExpr, stats *MapperSt
 	_, literalLHS := expr.LHS.(*parser.NumberLiteral)
 	_, literalRHS := expr.RHS.(*parser.NumberLiteral)
 	if literalLHS && literalRHS {
-		return expr, false, nil
+		return expr, false, fmt.Errorf("both operands of binary expression are number literals '%s'", expr)
```

Comment:
Why returning an error? We shouldn't error out, but just not splitting it.

Reply:
This was after comment: https://github.com/grafana/mimir/pull/2469#discussion_r929794502

We should not reach this point, as the `isSplittable` method already checks if both operands are literals: https://github.com/grafana/mimir/blob/06b6079ab1b7758229404721361c8439c733cf2c/pkg/frontend/querymiddleware/astmapper/instant_splitting.go#L118-L119

Therefore, the error is more to notify something is wrong as suggested.
But I'm ok, with removing the error also 🙂

Reply:
I'd rather keep the error since it's a sanity check here. If we reach this point, it's an error.